chain/fee_manager_test.go

```diff
@@ -0,0 +1,194 @@
+	bandwidthOp := math.NewUint64Operator(200)
+	bandwidthUnits, err := bandwidthOp.Value()
+	require.NoError(err)
+
+	computeUnitsOp := math.NewUint64Operator(rules.GetBaseComputeUnits())
+	computeUnitsOp.Add(50)
+	computeUnitsOp.Add(100)
+	computeUnits, err := computeUnitsOp.Value()
+	require.NoError(err)
+
+	readsOp := math.NewUint64Operator(0)
+	readsOp.Add(rules.GetStorageKeyReadUnits())
+	readsOp.MulAdd(uint64(chunks), rules.GetStorageValueReadUnits())
+	readUnits, err := readsOp.Value()
+	require.NoError(err)
+
+	allocatesOp := math.NewUint64Operator(0)
+	allocatesOp.Add(rules.GetStorageKeyAllocateUnits())
+	allocatesOp.MulAdd(uint64(chunks), rules.GetStorageValueAllocateUnits())
+	allocateUnits, err := allocatesOp.Value()
+	require.NoError(err)
+
+	writesOp := math.NewUint64Operator(0)
+	writesOp.Add(rules.GetStorageKeyWriteUnits())
+	writesOp.MulAdd(uint64(chunks), rules.GetStorageValueWriteUnits())
+	writeUnits, err := writesOp.Value()
+	require.NoError(err)
+
+	consumed := Dimensions{bandwidthUnits, computeUnits, readUnits, allocateUnits, writeUnits}
+
+	// Similar to processor.go
+	ok, dimension := feeManager.Consume(consumed, rules.GetMaxBlockUnits())
+	require.True(ok)
+	require.Equal(dimension, Dimension(0))
+
+	// The fee required 
+	_, err = feeManager.MaxFee(consumed)
+	require.NoError(err)
+
+	// Update last consumed
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		feeManager.SetLastConsumed(i, consumed[i])
+	}
+
+	unitsConsumed := feeManager.UnitsConsumed()
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		require.Equal(unitsConsumed[i], feeManager.LastConsumed(i))
+	}
+}
+
+func TestComputeNext(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Similar to genesis config
+	rules := NewMockRules(ctrl)
+	rules.EXPECT().GetWindowTargetUnits().Return(Dimensions{20_000_000, 1_000, 1_000, 1_000, 1_000})
+	rules.EXPECT().GetUnitPriceChangeDenominator().Return(Dimensions{48, 48, 48, 48, 48})
+	rules.EXPECT().GetMinUnitPrice().Return(Dimensions{100, 100, 100, 100, 100})
+
+	// Create our parent "block"
+	parentTime := time.Date(2024, time.January, 1, 0, 0, 0, 0, time.UTC).UnixMilli()
+	nextTime := time.Now().UnixMilli()
+
+	parentFeeManager := NewFeeManager(nil)
+	_, err := parentFeeManager.ComputeNext(parentTime, nextTime, rules)
+	require.NoError(err)
+}
+
+func TestAdd(t *testing.T) {
+	require := require.New(t)
+	// Similar to tokenvm/cmd/token-wallet/backend/backend.go
+	consumed := Dimensions{1,2,3,4,5}
+	nconsumed := Dimensions{}
+	tmpConsumed, err := Add(nconsumed, consumed)
+	require.NoError(err)
+	nconsumed = tmpConsumed
+	for i := 0; i < FeeDimensions; i++ {
+		require.Equal(nconsumed[i], consumed[i])
+	}
+}
+
+func TestDimensionGreater(t *testing.T) {
+	require := require.New(t)
+	d := Dimensions{2,3,4,5,6}
+	o := Dimensions{1,2,3,4,5}
+	result := d.Greater(o)
+	require.True(result)
+}
+
+func TestInvalidDimensionGreater(t *testing.T) {
```

Comment:
For this Valid/Invalid tests of Greater, you should use table-driven testing (here is a basic example): https://github.com/ava-labs/avalanchego/blob/310b77e221b8d0c5fcee3254900079b1987466f5/network/p2p/gossip/bloom_test.go#L18